### DISCUSSION TITLE
Alert not been show in SEB

### DISCUSSION THREAD

**Discussion by markdem** [SafeExamBrowser/seb-win-refactoring#563] [General]
We have an exam system that will time an exam out after 5 min of inactivity (no idea why..). If I run the exam with SEB 2.2 the following message gets displayed.

<img width="541" alt="Untitled" src="https://user-images.githubusercontent.com/84127023/217962989-1a4a5fd7-08f1-4f7c-9202-e235ac0c34b8.png">

We have recently updated the exam laptops we use and I installed the latest version of SEB (3.4.1) and now the message is not shown and the exam is closed when the timer runs out.
I have also tested this in Chrome (110.0.5481.78) and it also works fine. 

Does anyone have any idea why it would not be shown under SEB?

Thanks

**Comment by dbuechel** [comment]
Could you please post the log file of the browser component (to be found under `%LocalAppData%\SafeExamBrowser\Logs`). You may also want to try to enable the developer tools of the browser and check the web console while an exam times out to see if there are any errors or alike.

**Reply by markdem** [reply]
Sorry for the late reply.

The browser log only has the following 4 lines in it.

[1213/103348.910:ERROR:ssl_client_socket_impl.cc(982)] handshake failed; returned -1, SSL error code 1, net_error -101
[1213/103348.950:ERROR:ssl_client_socket_impl.cc(982)] handshake failed; returned -1, SSL error code 1, net_error -101
[1213/103349.863:ERROR:ssl_client_socket_impl.cc(982)] handshake failed; returned -1, SSL error code 1, net_error -101
[1213/103349.958:ERROR:ssl_client_socket_impl.cc(982)] handshake failed; returned -1, SSL error code 1, net_error -101

I have also checked the console, but no errors are coming up before, during or after the timeout.

**Reply by dbuechel** [reply]
I am not a browser engine developer, but these errors would indicate that there is an issue with the network or establishing a secure connection over it respectively.

**Reply by dbuechel** [reply]
Does your exam system work fine with any other Chromium-based browser? The difference between version 2.x and 3.x is exactly that we switched the browser engine from Firefox to Chromium.

**Reply by markdem** [reply]
Yes, I have also tested this in Chrome (110.0.5481.78) and it works fine. It is also the preferred browser by the vendor.

Is there anyway of turning on more logging to see what the SSL client is trying to connect to? The exam runs fine except for this one prompt and I don't see why it would need a separate connection for it.

Also, is there any easy way of switching back to firefox without uninstalling and reinstalling?

**Reply by dbuechel** [reply]
> Is there anyway of turning on more logging to see what the SSL client is trying to connect to?

You could try to run your exam in the CEF sample application, see https://cef-builds.spotifycdn.com/index.html. That way, we could determine whether it indeed is an issue with the browser engine itself or not.

>Also, is there any easy way of switching back to firefox without uninstalling and reinstalling?

No, SEB 3.x uses Chromium and likely won't ever revert to Firefox as browser engine.

**Reply by markdem** [reply]
I tried again using the CIF example app. Same thing, the pop up does not get shown..

I guess it is not a SEB issue in this case, but am wondering why it works fine in chrome?

Thanks for your help.

**Reply by dbuechel** [reply]
Yes, then it's definitely an issue with the browser engine. The reason is that CEF and Chrome are not exactly the same, they are both using the Chromium engine, but have a lot of different functionality around it.
